control-plane/go.mod

```diff
@@ -4,6 +4,8 @@ module github.com/hashicorp/consul-k8s/control-plane
 // We need to use a replace directive instead of directly pinning because `api` requires version `0.5.1` and will clobber the pin, but not the replace directive.
 replace github.com/hashicorp/consul/proto-public => github.com/hashicorp/consul/proto-public v0.1.2-0.20231128204611-fd1d97c3349d
 
+replace github.com/hashicorp/consul/sdk => github.com/hashicorp/consul/sdk v0.4.1-0.20231130173447-f44a9e1729df
```

Comment:
This points to my branch where I extended the Retry struct. When [this PR](https://github.com/hashicorp/consul/pull/19813) gets merged.